src/Opdex.Platform.Application.Abstractions/EntryQueries/MarketTokens/GetMarketTokenSnapshotsWithFilterQuery.cs

```diff
@@ -0,0 +1,35 @@
+using MediatR;
+using Opdex.Platform.Application.Abstractions.Models.Tokens;
+using Opdex.Platform.Common.Models;
+using Opdex.Platform.Infrastructure.Abstractions.Data.Queries;
+using System;
+
+namespace Opdex.Platform.Application.Abstractions.EntryQueries.MarketTokens
+{
+    /// <summary>
+    /// Retrieves snapshot data from a market for a given token.
+    /// </summary>
+    public class GetMarketTokenSnapshotsWithFilterQuery : IRequest<TokenSnapshotsDto>
+    {
+        /// <summary>
+        /// Creates a request to retrieve snapshot data from a market for a given token.
+        /// </summary>
+        /// <param name="token">The address of the token.</param>
+        /// <param name="market">The address of the market.</param>
+        /// <param name="cursor">The snapshot cursor filter.</param>
+        public GetMarketTokenSnapshotsWithFilterQuery(Address market, Address token, SnapshotCursor cursor)
+        {
+            if (market == Address.Empty) throw new ArgumentNullException(nameof(market));
+            if (token == Address.Empty) throw new ArgumentNullException(nameof(token));
+            if (cursor is null) throw new ArgumentNullException(nameof(cursor));
```

Comment:
Nit - as we create queries/handlers we should add tests for these validations w/ exception messages. 

Personally, I often small minor fixes necessary for these when writing tests whether the exception thrown is the wrong type, or the message I had was incorrect.

Reply:
IMO, testing isn't that important here and the exception message is unnecessary where we're null checking. The property name provides enough info and can be tested.  but I agree if we can we should test

Reply:
Added some tests and messages where we're checking Address == Empty

Reply:
Tests validating at least the property including in the exception is good enough for me, its more just about making sure the _right_ exception is thrown according to expectations.